gossipd/gossip.c

```diff
@@ -1769,11 +1769,14 @@ static struct io_plan *get_peers(struct io_conn *conn,
 	size_t n = 0;
 	struct pubkey *id = tal_arr(conn, struct pubkey, n);
 	struct wireaddr *wireaddr = tal_arr(conn, struct wireaddr, n);
+	const struct gossip_getnodes_entry **nodes;
 	struct pubkey *specific_id = NULL;
+	struct node_map_iter it;
 
 	if (!fromwire_gossip_getpeers_request(msg, msg, &specific_id))
 		master_badmsg(WIRE_GOSSIPCTL_PEER_ADDRHINT, msg);
 
+	nodes = tal_arr(conn, const struct gossip_getnodes_entry*, n);
```

Comment:
Start the size at `0`; `append_node` expands the array size by 1 each time you call it.  Or is this deliberate?

Reply:
It is deliberate because even the `id` and `wireaddr` arrays above are initialized with `n` elements.

Reply:
The `append_node` expands the array size and then loads to the newly-created space on the array.  You allocate `n` entries here that it seems to me never get loaded with data.  In addition, notice that you also send only `num` entries in `gossip_wire.csv`; if `num` is the same as `n` (plausible to me as you are sending `id` and `wireaddr` with the same `num`) then only the first `n` entries will get sent.

Reply:
Ahh yes, I understand now. Thanks!. Will change